torchtune/models/llama3_1/_position_embeddings.py

```diff
@@ -146,12 +147,15 @@ def forward(
             - s: sequence length
             - n_h: num heads
             - h_d: head dim
+
+        Raises:
+            RuntimeError: if RoPE cache is not initialized prior to forward call
         """
 
-        # TODO: remove once our distributed recipes are on FSDP2
         if not self.is_cache_built:
-            with torch.device(x.device):
-                self.rope_init()
+            raise RuntimeError(
+                "RoPE cache is not built. Please call rope_init() first."
```

Comment:
this feels a bit inconvenient. Do we have to raise the error?

I also think that we should update the other positional embedding to follow the same pattern. They dont have is_cache_built. In the recipes, we should add a check:

if hasattr(m, init_rope) and m.is_cache_bult

currently it is just:
if hasattr(m, init_rope)

Reply:
> this feels a bit inconvenient. Do we have to raise the error?

I think it's safest. Was talking with @joecummings about this a bit; previously we had an issue where RoPE was not getting initialized and so there were garbage values left over from the `to_empty` call. This was extremely hard to debug (Joe can tell you all about that one). 

> I also think that we should update the other positional embedding to follow the same pattern.

Updating the other one is fine but I might save that for now since it has a larger blast radius and isn't blocking anything like this is.

> if hasattr(m, init_rope) and m.is_cache_bult

I think it'd be `not m.is_cache_built`, right? Either way I don't think there's any real cost to calling it twice. But I plan to move this init logic into a common util soon, so we can make the change as part of that move.